go/vt/vtgate/planbuilder/ddl.go

```diff
@@ -17,6 +19,33 @@ const (
 	DifferentDestinations string = "Tables or Views specified in the query do not belong to the same destination"
 )
 
+type fkStrategy int
+
+const (
+	allow fkStrategy = iota
+	disallow
```

Comment:
The names `allow` and `disallow` seem to be too general for the scope of `planbuilder` package. I suggest renaming as `fkAllow` and `fkDisallow` or similar.